UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/operation/OperationContractUtil.java

```diff
@@ -134,32 +111,12 @@ public DetailedError getEmptyTransactionNameError() {
         return getUnprocessableEntityError(getEmptyInvalidParameter("transactionName"));
     }
 
-    public GenericError getApprovalDeniedError() {
-        return new GenericError()
-                .type("HLApprovalDenied")
-                .title("You are not allowed to approve the given operation");
-    }
-
-    public GenericError getRejectionDeniedError() {
-        return new GenericError()
-                .type("HLRejectionDenied")
-                .title("You are not allowed to reject the given operation");
-    }
-
-    public GenericError getApprovalImpossibleError(){
-        return new GenericError()
-                .type("HLApprovalImpossible")
-                .title("The operation is not in pending state");
-    }
-
-    public GenericError getRejectionImpossibleError(){
-        return new GenericError()
-                .type("HLRejectionImpossible")
-                .title("The operation is not in pending state");
-    }
+    public OperationData getOrInitializeOperationData(Context ctx, String initiator, String contractName, String transactionName, String params) throws ValidationError {
+        String clientId = getEnrollmentIdFromClientId(ctx.getClientIdentity().getId());
+        initiator = valueUnset(initiator) ? clientId : initiator;
 
-    public OperationData getOrInitializeOperationData(Context ctx, String initiator, String contractName, String transactionName, String params) throws NoSuchAlgorithmException {
-        String key = OperationContractUtil.getDraftKey(contractName, transactionName, params);
+        String key;
+        key = OperationContractUtil.getDraftKey(contractName, transactionName, params);
```

Comment:
```suggestion
        String key = OperationContractUtil.getDraftKey(contractName, transactionName, params);
```
no need to split declaration and assignment